### PR TITLE
Skip applying Identity for Lightning-Qubit

### PR DESCRIPTION
**Pull request by josephleekl**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [ ] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [ ] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**
Lightning qubit only accepts identity with 1 wires, which is not necessarily the case from Pennylane - especially from Catalyst.

**Description of the Change:**
Remove assert for wires not equal to 1 for Lightning Qubit applying identity gate.

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**
https://github.com/PennyLaneAI/catalyst/issues/1175

[sc-95544]

### CODE REVIEW

File: pennylane_lightning/core/simulators/lightning_qubit/gates/cpu_kernels/GateImplementationsLM.hpp [hunk 1]
```diff
@@ -1249,7 +1249,6 @@ class GateImplementationsLM : public PauliGenerator<GateImplementationsLM> {
                               const std::size_t num_qubits,
                               const std::vector<std::size_t> &wires,
                               [[maybe_unused]] const bool inverse) {
-        PL_ASSERT(wires.size() == 1);
```
Comment (josephleekl) [inline pennylane_lightning/core/simulators/lightning_qubit/gates/cpu_kernels/GateImplementationsLM.hpp:1252]: is it possible for no wires?

File: pennylane_lightning/core/simulators/lightning_qubit/catalyst/tests/Test_LightningSimulator.cpp [hunk 2]
```diff
@@ -118,6 +118,32 @@ TEST_CASE("LightningSimulator::GateSet", "[GateSet]") {
...
+        LQsim->State(view);
+
+        CHECK(state.at(0) == std::complex<double>{1, 0});
+
+        std::complex<double> sum{0, 0};
+        for (std::size_t ind = 1; ind < state.size(); ind++) {
+            sum += state[ind];
+        }
+
+        CHECK(sum == std::complex<double>{0, 0});
```
Comment (paul0403) [inline pennylane_lightning/core/simulators/lightning_qubit/catalyst/tests/Test_LightningSimulator.cpp:170]: This is a very creative way to check the zeros :laughing:

File: pennylane_lightning/core/simulators/lightning_qubit/gates/cpu_kernels/GateImplementationsLM.hpp (hunk 1 above)
Comment (paul0403) [reply pennylane_lightning/core/simulators/lightning_qubit/gates/cpu_kernels/GateImplementationsLM.hpp:1252]: This runs on deafult.qubit 
```python
@qml.qnode(qml.device("default.qubit", wires=0))
def hi():
    qml.Identity()
    return qml.state()

print(hi())
```

```
[1.+0.j]
```

File: pennylane_lightning/core/simulators/lightning_qubit/gates/tests/Test_GateImplementations_Nonparam.cpp [hunk 3]
```diff
@@ -108,6 +108,18 @@ void testApplyIdentity() {
...
+        auto st_pre = createZeroState<ComplexT>(num_qubits);
+        auto st_post = createZeroState<ComplexT>(num_qubits);
+        GateImplementation::applyHadamard(st_pre.data(), num_qubits, {index},
+                                          false);
+        GateImplementation::applyHadamard(st_post.data(), num_qubits, {index},
+                                          false);
+
+        GateImplementation::applyIdentity(st_pre.data(), num_qubits,
+                                          {index, index + 1}, false);
+        CHECK(std::equal(st_pre.begin(), st_pre.end(), st_post.begin()));
```
Comment (paul0403) [inline pennylane_lightning/core/simulators/lightning_qubit/gates/tests/Test_GateImplementations_Nonparam.cpp:138]: Two comments here:
- Should we add a new test for the identity gate on multiple wires instead of smuggling it in another test?
- Should we move the identity gate from single-qubit gate section to multi-qubit gate section? Both here in the test and in the actual code base.

Comment (josephleekl) [reply pennylane_lightning/core/simulators/lightning_qubit/gates/tests/Test_GateImplementations_Nonparam.cpp:138]: I've moved them around to reflect that it is not a single-qubit gate.

I think these tests are fine given the current structure of our tests

**Review by jzaia18 — APPROVED** [review]
All the changes here look good. Is there any plan to make the same changes for the other devices in a follow-on PR? It could be confusing having the different devices behave differently here

**Comment by josephleekl** [comment]
> All the changes here look good. Is there any plan to make the same changes for the other devices in a follow-on PR? It could be confusing having the different devices behave differently here

Thanks @jzaia18 - for the other devices we actually already do this (skip Identity), e.g. [LK](https://github.com/PennyLaneAI/pennylane-lightning/blob/d58285ac534e2370737dced47d199c4194a40408/pennylane_lightning/core/simulators/lightning_kokkos/StateVectorKokkos.hpp#L361) and [LG](https://github.com/PennyLaneAI/pennylane-lightning/blob/d58285ac534e2370737dced47d199c4194a40408/pennylane_lightning/core/simulators/lightning_gpu/StateVectorCudaManaged.hpp#L317)

File: pennylane_lightning/core/simulators/lightning_qubit/catalyst/tests/Test_LightningSimulator.cpp [hunk 4]
```diff
@@ -118,6 +118,32 @@ TEST_CASE("LightningSimulator::GateSet", "[GateSet]") {
         CHECK(sum == std::complex<double>{0, 0});
     }
 
+    SECTION("Identity gate - multiple wires") {
```
Comment (maliasadi) [inline pennylane_lightning/core/simulators/lightning_qubit/catalyst/tests/Test_LightningSimulator.cpp:147]: Could we also test this with zero number of wires? Is there a legit lowering for `qml.Identity()`?

File: .github/CHANGELOG.md [hunk 5]
```diff
@@ -14,6 +14,9 @@
 
 <h3>Internal changes ⚙️</h3>
 
+- Skip Identity operation in Lightning Qubit and removed assert for applying Identity gate with more than 1 wire.
```
Comment (maliasadi) [inline .github/CHANGELOG.md]: And zero-wire following [qml.Identity](https://docs.pennylane.ai/en/stable/_modules/pennylane/ops/identity.html#Identity) ?

File: pennylane_lightning/core/simulators/lightning_qubit/gates/tests/Test_GateImplementations_Nonparam.cpp [hunk 6]
```diff
@@ -108,9 +110,26 @@ void testApplyIdentity() {
                                           false);
         CHECK(std::equal(st_pre.begin(), st_pre.end(), st_post.begin()));
     }
+
+    // Apply Identity on 2 qubits
```
Comment (maliasadi) [inline pennylane_lightning/core/simulators/lightning_qubit/gates/tests/Test_GateImplementations_Nonparam.cpp:127]: Do we need to test this with zero-qubit?

**Review by maliasadi — COMMENTED** [review]
:raised_hands:

File: .github/CHANGELOG.md (hunk 5 above)
Comment (josephleekl) [reply .github/CHANGELOG.md]: updated!

File: pennylane_lightning/core/simulators/lightning_qubit/catalyst/tests/Test_LightningSimulator.cpp (hunk 4 above)
Comment (josephleekl) [reply pennylane_lightning/core/simulators/lightning_qubit/catalyst/tests/Test_LightningSimulator.cpp:147]: added test with 0 wires - In theory it should be possible to have that, but when, who knows...

File: .github/CHANGELOG.md [hunk 7]
```diff
@@ -14,7 +14,7 @@
 
 <h3>Internal changes ⚙️</h3>
 
-- Skip Identity operation in Lightning Qubit and removed assert for applying Identity gate with more than 1 wire.
+- Skip Identity operation in Lightning Qubit and removed assert for applying Identity gate not equal to 1 wire.
```
Comment (maliasadi) [inline .github/CHANGELOG.md]: minor: we could add this to "Improvements" as it extends the Identity gate op in LQ (C++). This will be required for https://github.com/PennyLaneAI/catalyst/pull/1907

**Review by maliasadi — APPROVED** [review]
Thanks @josephleekl! LGTM :)

File: .github/CHANGELOG.md (hunk 7 above)
Comment (josephleekl) [reply .github/CHANGELOG.md]: good point - I'll update it. Thanks!
